detect_secrets/pre_commit_hook.py

```diff
@@ -98,15 +131,45 @@ def raise_exception_if_baseline_file_is_not_up_to_date(filename):
         raise ValueError
 
     if filename.encode() in files_changed_but_not_staged:
-        CustomLog(formatter='%(message)s').getLogger()\
-            .error((
-                'Your baseline file ({}) is unstaged.\n'
-                '`git add {}` to fix this.'
-            ).format(
-                filename,
-                filename,
-            ))
+        _get_custom_log().error((
+            'Your baseline file ({}) is unstaged.\n'
+            '`git add {}` to fix this.'
+        ).format(
+            filename,
+            filename,
+        ))
+
+        raise ValueError
+
 
+def raise_exception_if_baseline_version_is_outdated(version):
+    """
+    Version changes may cause breaking changes with past baselines.
+    Due to this, we want to make sure that the version that the
+    baseline was created with is compatible with the current version
+    of the scanner.
+
+    We use semantic versioning, and check for bumps in the MINOR
+    version (a good compromise, so we can release patches for other
+    non-baseline-related issues, without having all our users
+    recreate their baselines again).
+
+    :type version: str|None
+    :param version: version of baseline
+    :raises: ValueError
+    """
+    if not version:
+        # Baselines created before this change, so by definition,
+        # would be outdated.
+        raise ValueError
+
+    baseline_version = version.split('.')
+    current_version = VERSION.split('.')
+
+    if int(current_version[0]) > int(baseline_version[0]):
+        raise ValueError
+    elif current_version[0] == baseline_version[0] and \
+            int(current_version[1]) > int(baseline_version[1]):
```

Comment:
So e.g. we make a baseline on `0.8.8`, and we auto-update `detect-secrets` via `all-repos` and the next time somebody runs `detect-secrets` version `0.9.0` we throw a value error? That seems good.

What does the pre-commit hook say to the user when the error is thrown?

Reply:
Ah whoops. Forgot to add error message. Too excited about that `lru_cache` =)